teacher.rb

```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(specialization:, age:, name: 'Unknown', parent_permission: true)
+    super(name: name, age: age, parent_permission: parent_permission)
```

Comment:
In order to conform to best practices of Ruby (ruby's hash omission syntax), kindly consider removing the hash values. In a hash where the key names and its variables are the same, you can just state the key name and omit the variables. For instance, instead of `super(name: name, age: age, parent_permission: parent_permission)`, you can use the shorthand syntax `super(name, age, parent_permission)`.
:stuck_out_tongue_winking_eye: